src/dmd/backend/cod3.d

```diff
@@ -208,17 +208,21 @@ private __gshared ubyte[256] inssize2 =
 ];
 
 /*************************************************
- * Allocate register temporaries
+ * Generate code to save `reg` in `regsave`E stack area.
+ * Params:
+ *      regsave = register save areay on stack
+ *      cdb = where to write generated code
+ *      reg = register to save
+ *      idx = set to location in regsave for use in REGSAVE_restore()
  */
 
-void REGSAVE_save(ref REGSAVE regsave, ref CodeBuilder cdb, reg_t reg, uint *pidx)
+void REGSAVE_save(ref REGSAVE regsave, ref CodeBuilder cdb, reg_t reg, out uint idx)
 {
-    uint i;
-    if (reg >= XMM0)
+    if (isXMMreg(reg))
```

Comment:
This will have slightly different behavior but it might be more correct.

Reply:
Not only is it more correct, it expresses the intent.